### DISCUSSION TITLE
IKEA E1743 wrong battery percentage

### DISCUSSION THREAD

**Discussion by a-bali** [arendst/Tasmota#20129] [Q&A]
I run Tasmota 13.2.0 on an Athom ZG01 and connected an IKEA E1743 switch. I had to upgrade the switch's firmware to the latest otherwise it paired but did not send any event. Now it works fine but the shown battery percentage (which it reports once per day) is wrong:

`2023-11-29T21:26:00+0100 tele/tasmota_680D40/SENSOR {"ZbReceived":{"0x3DFE":{"Device":"0x3DFE","Name":"switch","BatteryVoltage":2.9,"BatteryPercentage":180,"Endpoint":1,"LinkQuality":138}}}`

I suspect this might be because of the [default zigbee plugin](https://tasmota.github.io/docs/Zigbee/#complete-examples) built into the firmware which multiplies the percentage by 2.

Is it possible to override this rule from the default plugin without recompiling the firmware?

**Comment by s-hadinger** [comment]
Ha. Indeed, earlier Ikea had a bug and reported only half of the battery (IKEA's developers hadn't read carefully the spec)

You can override with a specific plugin in the file system, it will supersede the built-in one (at least it should)

**Reply by stefanbode** [reply]
Maybe we can make this automatic if the value is above 50, then we know THIS device now reports 0..100. So the device may be wrong first on low power but correct the first time it is charged
